### PR TITLE
BinaryData: raise instead of discarding data

### PR DESCRIPTION
**Pull request by cemsbr**
Before, packed BinaryData would be empty if given a non-byte value.
E.g.: ``BinaryData('string').pack() == b''``. Now, we raise ``ValueError`` to prevent the user from losing data.

### CODE REVIEW

File: pyof/foundation/basic_types.py [hunk 1]
```diff
@@ -411,8 +411,13 @@ def __init__(self, value=b''):  # noqa
         """The constructor takes the parameter below.
 
         Args:
-            value (bytes): The binary data. Defaults to an empty value.
+            value (bytes, str): The binary data. Defaults to an empty value.
```
Comment (diraol) [inline pyof/foundation/basic_types.py]: @cemsbr I think we do not accept `str` as a valid `value` right?

**Comment by cemsbr** [comment]
Documentation updated with @diraol's review.

File: pyof/foundation/basic_types.py [hunk 2]
```diff
@@ -430,8 +435,10 @@ def pack(self, value=None):
         if value is None:
```
Comment (renanrodrigo) [inline pyof/foundation/basic_types.py]: This line + line 438 could'n appear as 
`if not value:`
and then
`else`

or the other way around?

Comment (diraol) [inline pyof/foundation/basic_types.py]: I don't think so, because, in this specific case:
``None == 0 == b'' == ''`` (`==` -> behaves like)

Comment (renanrodrigo) [inline pyof/foundation/basic_types.py]: Indeed. Ok then (:

**Comment by diraol** [comment]
@cemsbr CHANGELOG?

**Pull request merged**